android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -216,34 +217,78 @@ public Object getSelectionKey()
 								}
 								return null;
 							}
+
+							@Override
+							public boolean inSelectionHotspot(@NonNull MotionEvent e)
+							{
+								// Returning true allows taps to immediately select this row.
+								return true;
+							}
 						};
 					}
 					return null;
 				}
 			},
 			StorageStrategy.createLongStorage()
-		)
-			.withSelectionPredicate(SelectionPredicates.<Long>createSelectSingleAnything())
-			.build();
-		this.tracker.addObserver(new SelectionTracker.SelectionObserver()
-		{
+		);
+
+		final KrollDict properties = proxy.getProperties();
+		final boolean editing = properties.optBoolean(TiC.PROPERTY_EDITING, false);
+		final boolean allowsSelection = properties.optBoolean(TiC.PROPERTY_ALLOWS_SELECTION_DURING_EDITING, false);
+		final boolean allowsMultipleSelection
+			= properties.optBoolean(TiC.PROPERTY_ALLOWS_MULTIPLE_SELECTION_DURING_EDITING, false);
+
+		if (editing && allowsSelection) {
+			if (allowsMultipleSelection) {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectAnything())
+					.build();
+			} else {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectSingleAnything())
+					.build();
+			}
 
-			@Override
-			public void onSelectionChanged()
-			{
-				super.onSelectionChanged();
+			if (this.tracker != null) {
+				this.tracker.addObserver(new SelectionTracker.SelectionObserver()
+				{
+					int selectionCount = 0;
+
+					@Override
+					public void onSelectionChanged()
+					{
+						super.onSelectionChanged();
+
+						selectionCount = tracker.hasSelection() ? tracker.getSelection().size() : 0;
+						selectedItems = new ArrayList<>(selectionCount);
 
-				/*if (tracker.hasSelection()) {
-					final Iterator<ListItemProxy> i = tracker.getSelection().iterator();
-					while (i.hasNext()) {
-						final ListItemProxy proxy = i.next();
+						if (tracker.hasSelection()) {
+							final Iterator<ListItemProxy> i = tracker.getSelection().iterator();
 
-						Log.d(TAG, "SELECTED: " + proxy.getProperties().getString(TiC.PROPERTY_TITLE));
+							while (i.hasNext()) {
+								final ListItemProxy item = i.next();
+								final KrollDict selectedItem = new KrollDict();
+								final ListSectionProxy section =
+									item.getParent() != null ? (ListSectionProxy) item.getParent() : null;
+
+								if (section != null) {
+									selectedItem.put(TiC.PROPERTY_ITEM_INDEX, item.getIndexInSection());
+									selectedItem.put(TiC.PROPERTY_SECTION, section);
+									selectedItem.put(TiC.PROPERTY_SECTION_INDEX, proxy.getIndexOfSection(section));
+
+									selectedItems.add(selectedItem);
+								}
+							}
+						}
+
+						final KrollDict data = new KrollDict();
+
+						data.put(TiC.PROPERTY_SELECTED_ITEMS, selectedItems.toArray(new KrollDict[0]));
+						data.put(TiC.PROPERTY_STARTING_ITEM, selectedItems.isEmpty() ? null : selectedItems.get(0));
+						proxy.fireEvent(TiC.EVENT_ITEMS_SELECTED, data);
```

Comment:
On iOS, when in single selection mode, only the "itemclick" event is fired. The "itemsselected" event is only fired when in multi-selection mode.
```java
if (allowsMultipleSelection) {
	final KrollDict data = new KrollDict();
	data.put(TiC.PROPERTY_SELECTED_ITEMS, selectedItems.toArray(new KrollDict[0]));
	data.put(TiC.PROPERTY_STARTING_ITEM, selectedItems.isEmpty() ? null : selectedItems.get(0));
	proxy.fireEvent(TiC.EVENT_ITEMS_SELECTED, data);
} else if (!selectedItems.isEmpty()) {
	selectedItems.get(0).fireSyncEvent(TiC.EVENT_CLICK, null);
}
```
_(I don't know if I'm handling the `EVENT_CLICK` line of code right so you might want to double check.)_